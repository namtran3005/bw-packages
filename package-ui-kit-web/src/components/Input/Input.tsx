import * as React from 'react';
import styled from 'styled-components';

import { Input as MuiInput } from '@material-ui/core';
import { InputProps as MuiInputProps } from '@material-ui/core/Input';

import { Colors, Transitions } from '@bitwala-cryptobank-squad/package-theme';

interface InputProps extends MuiInputProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const InputComponent: React.FunctionComponent<InputProps> = ({
  disableUnderline = true,
  ...props
}) => <MuiInput disableUnderline={disableUnderline} {...props} />;

const Input = styled(InputComponent).withConfig({
  shouldForwardProp: (prop) => !['textAlign'].includes(prop),
})`
  && {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: ${({ multiline }) => (multiline ? 'flex-start' : 'center')};
    padding: ${({ multiline }) => (multiline ? '12px 0px' : '')};
    padding-left: 12px;
    padding-right: 12px;
    border-radius: 4px;
    min-height: ${({ multiline }) => (multiline ? '128px' : '48px')};
    transition: ${Transitions.duration.quick}ms ${Transitions.easing.swiftOut};
    background: ${Colors.getWithAlpha('subtitleText', 0.1)};
    color: ${Colors.get('text')};
    border: 1px solid ${({ error }) => (error ? Colors.get('error') : 'white')};
    font-family: Inter;
    font-size: 0.875rem;
    font-weight: 500;
  }
  &:focus-within {
    background: ${Colors.get('white')};
    border: 1px solid
      ${({ error }) => Colors.get(error ? 'error' : 'primaryLilac')};
  }
`;

export { Input, InputProps };
