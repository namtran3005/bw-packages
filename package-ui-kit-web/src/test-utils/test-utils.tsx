import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
