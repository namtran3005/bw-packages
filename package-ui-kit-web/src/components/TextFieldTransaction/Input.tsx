import * as React from 'react';
import styled from 'styled-components';

import { Input as MuiInput } from '@material-ui/core';
import { InputProps as MuiInputProps } from '@material-ui/core/Input';

import { Colors, Transitions } from '@bitwala-cryptobank-squad/package-theme';

interface InputProps extends MuiInputProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const InputComponent: React.FunctionComponent<InputProps> = ({
  disableUnderline = true,
  ...props
}) => <MuiInput disableUnderline={disableUnderline} {...props} />;

const Input = styled(InputComponent).withConfig({
  shouldForwardProp: (prop) =>
    ![
      'textAlign',
      'startAdornmentLabel',
      'adornmentBlock',
      'maxButtonDisabled',
      'maxButtonOnClick',
      'maxButtonLabel',
      'maxButton',
    ].includes(prop),
})`
  && {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: ${({ multiline }) => (multiline ? 'flex-start' : 'center')};
    padding: ${({ multiline }) => (multiline ? '6px 0px' : '')};
    padding-left: 6px;
    margin: 6px 0px;
    border-radius: 50px;
    min-height: ${({ multiline }) => (multiline ? '128px' : '49px')};
    transition: ${Transitions.duration.quick}ms ${Transitions.easing.swiftOut};
    color: ${Colors.get('text')};
    border: 2px solid
      ${({ error }) =>
        error ? Colors.get('error') : Colors.get('backgroundLight')};
    font-family: Inter;
    font-size: 0.875rem;
    font-weight: 500;
    background-color: transparent;
  }
  &:focus-within {
    background: ${Colors.get('white')};
    border: 2px solid
      ${({ error }) =>
        error ? Colors.get('error') : Colors.get('primaryLilac')};
  }
`;

export { Input, InputProps };
