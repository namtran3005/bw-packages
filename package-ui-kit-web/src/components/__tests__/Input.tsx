import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { Input } from '../Input';

const defaultProps = {};

describe('Input', () => {
  it('should render input', () => {
    const { baseElement } = renderWithTestHarness(<Input {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
