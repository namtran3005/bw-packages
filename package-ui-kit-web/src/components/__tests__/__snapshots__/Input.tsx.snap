// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render input 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 12px;
  padding-right: 12px;
  border-radius: 4px;
  min-height: 48px;
  -webkit-transition: 150ms cubic-bezier(.62,.28,.23,.99);
  transition: 150ms cubic-bezier(.62,.28,.23,.99);
  background: rgba(109,109,134,0.1);
  color: #2C232E;
  border: 1px solid white;
  font-family: Inter;
  font-size: 0.875rem;
  font-weight: 500;
}

.c0:focus-within {
  background: #FFFFFF;
  border: 1px solid #BEAAFF;
}

<body>
  <div>
    <div
      class="MuiInputBase-root MuiInput-root c0"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        type="text"
        value=""
      />
    </div>
  </div>
</body>
`;
